Source/Scene/Cesium3DTile.js

```diff
@@ -0,0 +1,1018 @@
+         */
+        this.replacementNode = undefined;
+
+        var expire = header.expire;
+        var expireDuration;
+        var expireDate;
+        if (defined(expire)) {
+            expireDuration = expire.duration;
+            if (defined(expire.date)) {
+                expireDate = JulianDate.fromIso8601(expire.date);
+            }
+        }
+
+        /**
+         * The time in seconds after the tile's content is ready when the content expires and new content is requested.
+         *
+         * @type {Number}
+         * @readonly
+         */
+        this.expireDuration = expireDuration;
+
+        /**
+         * The date when the content expires and new content is requested.
+         *
+         * @type {JulianDate}
+         * @readonly
+         */
+        this.expireDate = expireDate;
+
+        /**
+         * Marks if the tile is selected this frame.
+         *
+         * @type {Boolean}
+         *
+         * @private
+         */
+        this.selected = false;
+
+
+        /**
+         * The time when a style was last applied to this tile.
+         *
+         * @type {Number}
+         *
+         * @private
+         */
+        this.lastStyleTime = 0;
+
+        /**
+         * Marks whether the tile's children bounds are fully contained within the tile's bounds
+         *
+         * @type {Cesium3DTileOptimizationHint}
+         *
+         * @private
+         */
+        this._optimChildrenWithinParent = Cesium3DTileOptimizationHint.NOT_COMPUTED;
+
+        // Members that are updated every frame for tree traversal and rendering optimizations:
+        this._distanceToCamera = 0;
+        this._visibilityPlaneMask = 0;
+        this._childrenVisibility = Cesium3DTileChildrenVisibility.VISIBLE;
+        this._lastSelectedFrameNumber = -1;
+        this._screenSpaceError = 0;
+        this._screenSpaceErrorComputedFrame = -1;
+        this._finalResolution = true;
+        this._requestHeap = undefined;
+        this._depth = 0;
+        this._centerZDepth = 0;
+        this._stackLength = 0;
+        this._selectedFrame = -1;
+        this._selectionDepth = 0;
+        this._lastSelectionDepth = undefined;
+        this._requestedFrame = undefined;
+        this._lastVisitedFrame = undefined;
+        this._ancestorWithContent = undefined;
+        this._ancestorWithLoadedContent = undefined;
+
+        this._debugBoundingVolume = undefined;
+        this._debugContentBoundingVolume = undefined;
+        this._debugViewerRequestVolume = undefined;
+        this._debugColor = Color.fromRandom({ alpha : 1.0 });
+        this._debugColorizeTiles = false;
+
+        this._commandsLength = 0;
+
+        this._color = undefined;
+        this._colorDirty = false;
+    }
+
+    defineProperties(Cesium3DTile.prototype, {
+        /**
+         * The tileset containing this tile.
+         *
+         * @memberof Cesium3DTile.prototype
+         *
+         * @type {Cesium3DTileset}
+         * @readonly
```

Comment:
I think a bunch of these cases are properties that are readonly from an API perspective but are able to be set from other 3D Tiles-related files.